internal/controller/apiextensions/composite/composition_render.go

```diff
@@ -61,13 +61,16 @@ func RenderFromJSON(o resource.Object, data []byte) error {
 	o.SetName(name)
 	o.SetNamespace(namespace)
 
-	// This resource already had a GVK (probably because it already exists), but
+	// This resource already had a GK (probably because it already exists), but
 	// when we rendered its template it changed. This shouldn't happen. Either
-	// someone changed the kind in the template or we're trying to use the wrong
-	// template (e.g. because the order of an array of anonymous templates
+	// someone changed the kind or group in the template, or we're trying to use the
+	// wrong template (e.g. because the order of an array of anonymous templates
 	// changed).
-	if !gvk.Empty() && o.GetObjectKind().GroupVersionKind() != gvk {
-		return errors.Errorf(errFmtKindChanged, gvk, o.GetObjectKind().GroupVersionKind())
+	// Please note, we don't check for version changes, as versions can change. For example,
+	// if a composed resource was created with a template that has a version of "v1alpha1",
+	// and then the template is updated to "v1beta1", the composed resource will still be valid.
+	if !gvk.Empty() && o.GetObjectKind().GroupVersionKind().GroupKind() != gvk.GroupKind() {
```

Comment:
I wonder if we could go back to only checking "Kind." I could see checking the group backfiring during future migrations.

Reply:
I am not sure, if a migration requires a change in `Group`, it could very well require a change in `Kind` as well. `Version` differentiates here. 
I would imagine a migration tool updating references in such a case, rather than trying to accept here.

Reply:
yeah, sounds reasonable, I was just considering it given that it was the original behaviour